src/importexport/guitarpro/internal/guitarbendimport/benddatacollector.cpp

```diff
@@ -0,0 +1,286 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "benddatacollector.h"
+
+#include <engraving/dom/chord.h>
+#include <engraving/dom/note.h>
+#include <engraving/dom/score.h>
+#include <engraving/dom/tie.h>
+
+using namespace mu::engraving;
+
+namespace mu::iex::guitarpro {
+static void fillBendDataContextForNote(BendDataContext& bendDataCtx, const BendDataCollector::ImportedBendInfo& importedInfo);
+
+void BendDataCollector::storeBendData(mu::engraving::Note* note, const mu::engraving::PitchValues& pitchValues)
+{
+    if (!pitchValues.empty()) {
+        m_bendInfoForNote[note->track()][note->tick().ticks()] = fillBendInfo(note, pitchValues);
+    }
+}
+
+BendDataContext BendDataCollector::collectBendDataContext()
+{
+    BendDataContext bendDataCtx;
+
+    for (const auto& [track, trackInfo] : m_bendInfoForNote) {
+        for (const auto& [tick, importedBendInfo] : trackInfo) {
+            fillBendDataContextForNote(bendDataCtx, importedBendInfo);
+        }
+    }
+
+    return bendDataCtx;
+}
+
+BendDataCollector::ImportedBendInfo BendDataCollector::fillBendInfo(Note* note, const PitchValues& pitchValues)
+{
+    PitchValues adaptedPitchValues;
+
+    if (pitchValues.front().time != 0) {
+        PitchValue firstPv = pitchValues.front();
+        firstPv.time = 0;
+        adaptedPitchValues.push_back(firstPv);
+    }
+
+    for (const auto& pv : pitchValues) {
+        adaptedPitchValues.push_back(pv);
```

Comment:
I'd reserve pitchValues.size() in the line 56. Otherwise for 3 values you have 3 allocations and 2 deallocations here.
I'd do that everywhere where the size is known.